### PR TITLE
bug 1871109: Add a user friendly error message in VM import wizard

### PR DESCRIPTION
**Pull request by irosenzw**
Signed-off-by: Ido Rosenzwig <irosenzw@redhat.com>

When non-admin user picks a provider to import a VM from and lacks the permissions to do so.
minimized_error

The error expended :
maximaized_error

### CODE REVIEW

**Comment by matthewcarleton** [comment]
How about this for the content?
Screen Shot 2020-08-25 at 2 15 59 PM
I guess all our alerts surface at the bottom of the wizard right? 
I wonder if we should also show an error on the provider dropdown to?
"Insufficient permissions" WDYT?

**Comment by irosenzw** [comment]
The error message suppose to be generic so it will suites other pages as well (like creating a VM), not just VM import.

**Comment by matthewcarleton** [comment]
Does it have to be generic? If so, we could drop the part about importing but leave the rest.

**Comment by irosenzw** [comment]
@matthewcarleton yeah, I think that can work - I do like the `insufficient permissions` title.
Also, can we drop "or view more details below" part at the end?
It seems redundant since the user can do it in addition to contacting the administrator and most of the time he will view the details even if we won't instruct him to do so.

**Comment by irosenzw** [comment]
@matthewcarleton, updated error message:
Screenshot from 2020-08-26 11-36-51

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/error-components/utils.ts [hunk 1]
```diff
@@ -163,3 +167,30 @@ export const computeStorageErrors = (state, wizardReduxID) => {
 
   return { hasAllRequiredFilled, isValid, errors };
 };
+
+const isPermissionError = (title: string, message: string): boolean =>
+  title.includes('roles.rbac.authorization.k8s.io') ||
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/error-components/utils.ts:172]: this is only a specific error for operating on roles. IMO we should check the http code and mention 403 all errors.

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/error-components/utils.ts [hunk 2]
```diff
@@ -163,3 +167,30 @@ export const computeStorageErrors = (state, wizardReduxID) => {
...
+
+  if (isPermissionError(errTitle, message)) {
+    return {
+      isLong: true,
+      errTitle: INSUFFICIENT_PERMISSIONS_ERROR_TITLE,
+      description: INSUFFICIENT_PERMISSIONS_ERROR_DESC,
+    };
+  }
+
+  const [desc] = message.split('\n');
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/error-components/utils.ts:194]: what is the use case? in what conditions does this occur?

**Review by suomiy — CHANGES_REQUESTED** [review]

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/error-components/utils.ts (hunk 1 above)
Comment (irosenzw) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/error-components/utils.ts:172]: agree.

**Comment by matthewcarleton** [comment]
> @matthewcarleton yeah, I think that can work - I do like the `insufficient permissions` title.
> Also, can we drop "or view more details below" part at the end?
> It seems redundant since the user can do it in addition to contacting the administrator and most of the time he will view the details even if we won't instruct him to do so.

ya I think it's ok to drop that part of the text and just have the View details action.

**Comment by suomiy** [comment]
this bug/PR is fixed and superseeded by #6480

**Pull request merged**